tests/unittests/net/test_net_rendering.py

```diff
@@ -88,6 +88,9 @@ def _check_network_manager(network_state: NetworkState, tmp_path: Path):
     "test_name, renderers",
     [("no_matching_mac_v2", Renderer.Netplan | Renderer.NetworkManager)],
 )
+@pytest.mark.xfail(
+    reason="v2 interface-specific DNS errantly gets applied globally"
```

Comment:
Does this still fail with the new changes?

Reply:
Yes, because a subnet-specific DNS entry will get added to the global list, so the global entry is getting added to connections that didn't specify any DNS. I'll follow up on this separately.